src/contracts/atlas/Escrow.sol

```diff
@@ -177,15 +180,17 @@ abstract contract Escrow is AtlETH {
         // Set the gas baseline
         uint256 gasWaterMark = gasleft();
 
-        EscrowAccountBalance memory solverEscrow = _balanceOf[solverOp.from];
+        EscrowAccountAccessData memory aData = accessData[solverOp.from];
 
-        uint256 solverBalance = uint256(solverEscrow.total - solverEscrow.bonded);
+        uint256 solverBalance = aData.bonded;
+        uint256 lastAccessedBlock = aData.lastAccessedBlock;
 
         if (solverOp.to != address(this)) {
             result |= 1 << uint256(SolverOutcome.InvalidTo);
         }
 
-        if (accessData[solverOp.from].lastAccessedBlock >= uint64(block.number)) {
+        // NOTE: Turn this into time stamp check for FCFS L2s?
```

Comment:
Timestamp checking may also be better for deterministic unbonding wait times, on chains where the time between blocks varies.

Reply:
True! i think the only concern is that timestamp might not be reliable on some chains. But i think either block.number or timestamp should be reliable - it just depends where we are to know which.